### PR TITLE
Review date

### PR DESCRIPTION
**Pull request by sabi98**

### CODE REVIEW

**Comment by sabi98** [comment]
@mikemonteith Tests passed. :heavy_check_mark:

File: docs/components/review_date.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+# Review Date
+
+Use this code snippet to include the review date component. This snipped has to be pasted in `home/models.py' and then added to the page through the Wagtail admin interface.
```
Comment (mikemonteith) [inline docs/components/review_date.md]: mismatched backtick and aopstrophie: `home/models.py'

File: docs/components/review_date.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+
+To include the review date component on the page you will have to add this snippet of code in the template:
+
+    {% include "wagtailnhsukfrontend/review_date.html" %}
```
Comment (mikemonteith) [inline docs/components/review_date.md]: Put this inside a code block

File: docs/components/review_date.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```py
+from wagtail.admin.edit_handlers import StreamFieldPanel
+from wagtail.core.models import Page
+from wagtail.core.fields import StreamField
+from wagtailnhsukfrontend.mixins import (
+ReviewDate,
+)
+ 
+class  HomePage(Page, ReviewDate):
+parent_page_types = ['wagtailcore.Page']
```
Comment (mikemonteith) [inline docs/components/review_date.md]: some indentation issues with the python code.

File: testapp/home/migrations/0005_auto_20190307_1616.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Generated by Django 2.0.13 on 2019-03-07 16:16
```
Comment (mikemonteith) [inline testapp/home/migrations/0005_auto_20190307_1616.py]: Can you give this migration a nicer name?
It needs to start `0005_` but could be called `0005_add_review_dates.py` or something

File: testapp/home/models.py [hunk 5]
```diff
@@ -2,6 +2,10 @@
 from wagtail.core.models import Page
 from wagtail.core.fields import StreamField
 
+from wagtailnhsukfrontend.mixins import (
+    ReviewDate,
```
Comment (mikemonteith) [inline testapp/home/models.py]: I think this class needs to be called `ReviewDateMixin` instead of just `ReviewDate`.

It's a wagtail convention.

File: testapp/home/models.py [hunk 6]
```diff
@@ -33,6 +37,11 @@ class HomePage(Page):
         StreamFieldPanel('body'),
     ]
 
+    settings_panels = Page.settings_panels + ReviewDate.settings_panels + [
```
Comment (mikemonteith) [inline testapp/home/models.py]: Don't need to add an empty array to the end of this line.
Could just be `settings_panels = Page.settings_panels + ReviewDate.settings_panels`

File: testapp/home/templates/home/home_page.html [hunk 7]
```diff
@@ -1,7 +1,12 @@
...
-
 {% block body_class %}template-homepage{% endblock %}
 
 {% block content %}
   {{ page.body }}
+
+  {% include "wagtailnhsukfrontend/review_date.html" %}
+ 
 {% endblock %}
+
```
Comment (mikemonteith) [inline testapp/home/templates/home/home_page.html]: Unnecessary whitespace additions here

File: testapp/home/templates/home/home_page.html [hunk 8]
```diff
@@ -1,7 +1,12 @@
 {% extends "base.html" %}
-
```
Comment (mikemonteith) [inline testapp/home/templates/home/home_page.html]: Unnecessary whitespace removal here

File: docs/components/review_date.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+StreamFieldPanel('body'),
+
+]
+
+settings_panels = Page.settings_panels + ReviewDate.settings_panels + [
+
+]
+```
+
+You can modify the componet settings in the page settings tab on the Wagtail admin interface.
```
Comment (mikemonteith) [inline docs/components/review_date.md]: `componet`!

**Comment by mikemonteith** [comment]
I've put comments on individual changes.
You will also need to squash the branch down into 1 commit.

File: wagtailnhsukfrontend/mixins.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+class ReviewDateMixin(models.Model):
+
+    last_review_label = models.CharField(verbose_name="Reviewed date label", max_length=120, help_text="The default for NHS frontend is: Page last reviewed:")
+    reviewed_date = models.DateTimeField(null=True, help_text="This will be displayed as Month/Year")
+    next_review_label = models.CharField(blank=True, null=True, max_length=120, help_text="The default for NHS frontend is: Page last reviewed: - Leave this empty if you do not want to add Next review date")
+    next_review_date = models.DateTimeField(blank=True, null=True, help_text="This will be displayed as Month/Year")
+
+    settings_panels = [
+        MultiFieldPanel(
```
Comment (mikemonteith) [inline wagtailnhsukfrontend/mixins.py]: MultiFieldPanel needs a `heading="Review Dates"` or something so it shows up nicely in the admin.

File: wagtailnhsukfrontend/mixins.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
+from django.db import models
+from wagtail.admin.edit_handlers import MultiFieldPanel, FieldPanel
+
+
+class ReviewDateMixin(models.Model):
+
+    last_review_label = models.CharField(verbose_name="Reviewed date label", max_length=120, help_text="The default for NHS frontend is: Page last reviewed:")
```
Comment (mikemonteith) [inline wagtailnhsukfrontend/mixins.py]: Need to have a rethink about these labels. I don't think the label should be a field on every page. More likely users will want to define the label once, not have to repeat it on every new page they create.
(I'm not sure how to actually do this yet, need to have a think about it).
